src/figuresClasses.ts

```diff
@@ -1,11 +1,95 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Figure['color'];
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Figure['color'], a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('This figure does not have a negative value');
```

Comment:
The error message 'This figure does not have a negative value' is misleading. It should be something like 'Side lengths must be positive' to accurately describe the error condition.